packages/extras/src/lib/components/Billboard/Billboard.svelte

```diff
@@ -0,0 +1,50 @@
+<script lang="ts">
+	import { Group, Quaternion } from 'three';
+	import { T, useTask, useThrelte } from '@threlte/core';
+	import type { BillboardEvents, BillboardProps, BillboardSlots } from './Billboard.svelte';
+
+	type $$Props = Required<BillboardProps>;
+	type $$Events = BillboardEvents;
+	type $$Slots = BillboardSlots;
+
+	export let follow: $$Props['follow'] = true;
+	export let lockX: $$Props['lockX'] = false;
+	export let lockY: $$Props['lockY'] = false;
+	export let lockZ: $$Props['lockZ'] = false;
+
+	let inner: Group;
+	let localRef: Group;
+
+	const { camera } = useThrelte();
+
+	const q = new Quaternion();
+
+	useTask(() => {
+		if (!follow || !localRef) return;
+
+		// save previous rotation in case we're locking an axis
+		const prevRotation = localRef.rotation.clone();
```

Comment:
We can improve garbage collection pressure a bit by reusing an euler here:

```
const prevRotation = new Euler()

useTask(() => {
  ...
  prevRotation.copy(localRef.rotation)
  ...
})

Reply:
Thanks! Changed